### PR TITLE
feat: add support for roles

### PR DESCRIPTION
**Pull request by tjrivera**
This PR adds the following:

* Support for operations on the `Roles` endpoint
* Support for Roles related operations in relation to `RBACUsers`

### CODE REVIEW

**Comment by hbagdi** [comment]
Should these RBAC roles not be scoped to a workspace?

**Comment by tjrivera** [comment]
I believe that RBAC roles _should_ in practice be scoped to a workspace but that it's not required. Similar to other endpoints throughout, this endpoint relies on client configuration of the workspace to drive scope.

File: kong/kong.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ type Client struct {
 	Workspaces     *WorkspaceService
 	Admins         *AdminService
 	RBACUsers      *RBACUserService
+	Roles          *RoleService
```
Comment (hbagdi) [inline kong/kong.go:47]: Please rename this to RBACRoles.

File: kong/types.go [hunk 2]
```diff
@@ -258,3 +258,12 @@ type WorkspaceEntity struct {
 	WorkspaceID      *string `json:"workspace_id,omitempty" yaml:"workspace_id,omitempty"`
 	WorkspaceName    *string `json:"workspace_name,omitempty" yaml:"workspace_name,omitempty"`
 }
+
+// Role represents an RBAC Role in Kong.
+type Role struct {
```
Comment (hbagdi) [inline kong/types.go:263]: Please rename this to RBACRole.

File: kong/types.go [hunk 3]
```diff
@@ -258,3 +258,12 @@ type WorkspaceEntity struct {
 	WorkspaceID      *string `json:"workspace_id,omitempty" yaml:"workspace_id,omitempty"`
 	WorkspaceName    *string `json:"workspace_name,omitempty" yaml:"workspace_name,omitempty"`
 }
+
+// Role represents an RBAC Role in Kong.
```
Comment (hbagdi) [inline kong/types.go:262]: The comment needs to be updated now.

File: kong/role_service.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// RoleService handles Roles in Kong.
+type RoleService service
+
+// Create creates a Role in Kong.
+func (s *RoleService) Create(ctx context.Context,
```
Comment (hbagdi) [inline kong/role_service.go:14]: This should be RBACRoleService.

**Review by hbagdi — APPROVED** [review]

**Pull request merged**